### PR TITLE
Enabled third party Mellanox TensorFlow CI

### PR DESCRIPTION
**Pull request by artemry-nv**
See added README section for details about additional on demand TensorFlow CI by Mellanox.

UPD (for those who received the comment "Can one of the admins verify this patch?" from mellanox-github user ): please just ignore it (seems some specific behaviour of the Github Jenkins plugin). We're investigating how to eliminate this. Sorry for disturbance.

### CODE REVIEW

**Comment by itemko** [comment]
@googlebot I signed it!

**Comment by itemko** [comment]
bot:mlx:test

**Comment by itemko** [comment]
@googlebot I fixed it.

**Comment by jackwish** [comment]
Is comments like https://github.com/tensorflow/tensorflow/pull/31791#issuecomment-523004009 is introduced by Mellanox tools?

**Comment by itemko** [comment]
> Is comments like [#31791 (comment)](https://github.com/tensorflow/tensorflow/pull/31791#issuecomment-523004009) is introduced by Mellanox tools?

It looks like - let me double check.

**Comment by itemko** [comment]
> Is comments like [#31791 (comment)](https://github.com/tensorflow/tensorflow/pull/31791#issuecomment-523004009) is introduced by Mellanox tools?

Please just ignore it, sorry for disturbance.

**Comment by jackwish** [comment]
> > Is comments like [#31791 (comment)](https://github.com/tensorflow/tensorflow/pull/31791#issuecomment-523004009) is introduced by Mellanox tools?
> 
> It looks like - let me double check.

Thank you. I notice that many opened PRs have received this comment, could be identified as spam :)

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 27-Aug-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/d7fb66960f2b203fc7285d22b80cffc9956c2bd8/build_nccl.log) | :heavy_check_mark: PASS
[build_tf.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/1f4c59b3e283043edee042c6924dc135b57d667b/build_tf.log) | :heavy_check_mark: PASS
[build_tf_launcher.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/bb0194f001ccb1a5703d299db99618cd5906aff8/build_tf_launcher.log) | :heavy_check_mark: PASS
[ml-tf-test-launcher.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/5988e8f93f250e3ce9d0567a725b25e56e64e1b9/ml-tf-test-launcher.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/acc1c2a135172a811f4c3093faa88a8c0ee453dc/run_nccl_test.log) | :heavy_check_mark: PASS
[run_tf_cnn_benchmarks.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/66c974818d16c18144ee6e964e33cc0507447576/run_tf_cnn_benchmarks.log) | :heavy_check_mark: PASS
[run_tf_models_imagenet_host1.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/7b8650aa766f9347e1782411f1c7eb034590adf5/run_tf_models_imagenet_host1.log) | :heavy_check_mark: PASS
[run_tf_models_imagenet_host2.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/4157b834e79e06ba6503639ccbce472e30fca44c/run_tf_models_imagenet_host2.log) | :heavy_check_mark: PASS
[run_tf_models_imagenet_launcher.log](https://gist.githubusercontent.com/mellanox-github/c80ad34ec72fd7f349378045b8bf5a4b/raw/061d528697775e8d28a9323f3e1812016067388f/run_tf_models_imagenet_launcher.log) | :heavy_check_mark: PASS

**Comment by martinwicke** [comment]
Please disable the commenting. We can merge this after.

File: tensorflow/tools/ci_build/README.md
Comment (jackwish) [inline tensorflow/tools/ci_build/README.md]: Would it be better that the tools is triggered ONLY when a bot were AT, say `@bot bot:mlx:test`? In this way, there is no need to keep scanning the PRs.

Comment (itemko) [reply tensorflow/tools/ci_build/README.md]: Could you please a bit elaborate on this? Do you mean changing polling logic (track PR creation/changes in the PR comments) into other mechanisms? Webhooks?

**Comment by itemko** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 02-Sep-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/360e820ac3f3b2aa9e5e93bee5e381a046752bb5/build_nccl.log) | :heavy_check_mark: PASS
[build_tf.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/5ed0927d11f4964c8e7c2bbc771082d3e51c5883/build_tf.log) | :heavy_check_mark: PASS
[build_tf_launcher.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/80b69d291946984a6deef896cfe6a5424a855ea3/build_tf_launcher.log) | :heavy_check_mark: PASS
[ml-tf-test-launcher.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/67fd8fa34bd90259ac2dfad461b0785e2e2e4bf5/ml-tf-test-launcher.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/60b3755927bc70e4539fe6046b8f69a02d833514/run_nccl_test.log) | :heavy_check_mark: PASS
[run_tf_cnn_benchmarks.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/5b0f3fc58fcaaa31b1f90bdeff6b9ae9fc460e51/run_tf_cnn_benchmarks.log) | :heavy_check_mark: PASS
[run_tf_models_imagenet_host1.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/9a3c4260ae0a7b5aa543f753fdb7ca7ca8d0ea6b/run_tf_models_imagenet_host1.log) | :heavy_check_mark: PASS
[run_tf_models_imagenet_host2.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/7ebc3062fb593ec5f1179e27abf382ce0c281035/run_tf_models_imagenet_host2.log) | :heavy_check_mark: PASS
[run_tf_models_imagenet_launcher.log](https://gist.githubusercontent.com/mellanox-github/cd15dc0910674bf1bc977b609f10782b/raw/39e91fbe53d9436af89a90fd332e7e74fcdef917/run_tf_models_imagenet_launcher.log) | :heavy_check_mark: PASS

**Comment by itemko** [comment]
@martinwicke 
> Please disable the commenting. We can merge this after.

Extra-commenting has been disabled - now any PR author can trigger Mellanox TensorFlow CI (without an approval) with the comment **bot:mlx:test**
Only [CI status](https://github.com/tensorflow/tensorflow/pull/31792#issuecomment-524773779) and [links to the artefacts](https://github.com/tensorflow/tensorflow/pull/31792#issuecomment-524773940) (log files) will be reported through the comments.

**Review by martinwicke — APPROVED** [review]

**Comment by itemko** [comment]
@martinwicke 
Submitted PR #32011 for master.

**Pull request closed without merging**
